content/docs/concepts/components_description.md

```diff
@@ -0,0 +1,202 @@
+
+## Admission Webhook k8s
+Admission Webhook K8s simplifies working with NSM if NSM is deployed in a Kubernetes cluster. This component automatically injects NSM clients into Kubernetes pods with NSM annotations. It can also inject NSM clients into entire pod namespaces.
+
+**Image**: https://github.com/networkservicemesh/cmd-admission-webhook-k8s/pkgs/container/cmd-admission-webhook-k8s
+
+**Responsibilities**:
+1. Injects NSM Clients into Kubernetes pods or namespaces if they have an NSM annotation
+
+## Cluster Info K8s
+**Cluster Info K8s** allows NSM to work with cluster properties: https://github.com/kubernetes-sigs/about-api. This component can be used to simplify retrieval of the cluster domain name for **Network Service Manager Proxy**.
+
+**Image**: https://github.com/networkservicemesh/cmd-cluster-info-k8s/pkgs/container/cmd-cluster-info-k8s
+
+**Responsibilities**:
+1. Obtains and updates information about the Kubernetes Cluster
+
+## Exclude Prefixes K8s
+**Exclude Prefixes K8s** collects prefixes from various sources in the Kubernetes Cluster including pods, services and config maps. These prefixes are used by NSM IPAM to avoid the intersection of prefixes of NSM and Kubernetes networks.
+
+**Image**: https://github.com/networkservicemesh/cmd-exclude-prefixes-k8s/pkgs/container/cmd-exclude-prefixes-k8s
+
+**Responsibilities**:
+1. Obtains and updates prefixes used in the Kubernetes Cluster
+
+## Map IP K8s
+**Map IP K8s** builds and updates a map that transfers internal pods' IPs of the Kubernetes Cluster to the external nodes' IPs. This map is used by **Network Service Manager Proxy** for interdomain NSM scenarios when NSM is deployed in the Kubernetes Cluster.
+
+**Image**: https://github.com/networkservicemesh/cmd-map-ip-k8s/pkgs/container/cmd-map-ip-k8s
+
+**Responsibilities**:
+1.  Builds and updates a map that transfers internal pods' IPs to external nodes' IPs
+
+## Network Service Endpoint (NSE)
+**Network Service Endpoint** provides NSM clients with access to **Network Services**.
+
+**Image**: https://github.com/networkservicemesh/cmd-nse-icmp-responder/pkgs/container/cmd-nse-icmp-responder
+
+**Implements services**:
+1. NetworkService
+
+**Responsibilities**:
+1. Provides NSM clients with access to **Network Services**
+2. Fills in remaining connection properties like IP, hardware addresses, and DNS configurations for the user's goals.
+
+## Network Service Client (NSC)
+**Network Service Client** allows external workloads to request access to NSM's **Network Services** and maintains connections ([Controlplane](https://networkservicemesh.io/docs/concepts/features/healing) and [Dataplane](https://networkservicemesh.io/docs/concepts/features/datapath_healing) healing). This component can also provide a local DNS Server for accessing NSM resources by their names. Supports only **kernel** and **vfio** mechanisms.
```

Comment:
I wouldn't point out that it only supports kernel and vfio, because nsc-memif is also NSC.
We can add a link to NSC-memif below (Image -> Repositories)